.claude/skills/layer5-blog-writer/scripts/generate_hero_image.py

```
@@ -428,63 +609,121 @@ def generate_hero_svg(title, subtitle, category, five_pose_arg, sign_text,
+    svg_path.write_text(svg_content, encoding="utf-8")
+
+    try:
+        detail = rasterize(svg_path, out, img_width, img_height, quality)
+    except RasterizeError as exc:
+        kept = out.with_suffix(".svg")
+        if not keep_svg:
+            kept.write_text(svg_content, encoding="utf-8")
+        raise SystemExit(
+            f"error: could not rasterize the hero image.\n  {exc}\n"
+            f"  The working SVG was left at {kept} so no work is lost."
+        )
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)